src/cpp/src/whisper_pipeline_base.hpp

```diff
@@ -0,0 +1,36 @@
+// Copyright (C) 2024 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+
+#pragma once
+
+#include "openvino/genai/whisper_pipeline.hpp"
+#include "openvino/genai/streamer_base.hpp"
+
+#include "whisper/whisper_feature_extractor.hpp"
+
+namespace ov {
+namespace genai {
+
+class WhisperPipelineImplBase {
+public:
+    WhisperPipelineImplBase(const WhisperGenerationConfig& config,
+                            const Tokenizer& tokenizer,
+                            const WhisperFeatureExtractor& feature_extractor)
+    : m_generation_config(config), m_tokenizer(tokenizer), m_feature_extractor(feature_extractor) {
+    }
+
+    virtual DecodedResults generate(const RawSpeechInput& raw_speech_input,
+                                    OptionalWhisperGenerationConfig generation_config,
+                                    StreamerVariant streamer) = 0;
+
+    virtual ~WhisperPipelineImplBase() = default;
+
+    WhisperGenerationConfig m_generation_config;
```

Comment:
move to protected section all the fields?

Reply:
Fields are being accessed like this: https://github.com/openvinotoolkit/openvino.genai/blob/abbc695715a1d0a92facd807fe63c6d2088b99e7/src/cpp/src/whisper_pipeline.cpp#L114

In that case, it might require introducing "getters". I kept it like this to minimize the changes.